src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,70 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { TodoStatus } from '../../types/TodoStatus';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  selectFilter: TodoStatus,
+  setSelectFilter: (sortStatus: TodoStatus) => void
+  setFilterField: (e: string) => void;
+  filterField: string,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  setSelectFilter,
+  selectFilter,
+  setFilterField,
+  filterField,
+}) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            className="is-capitalized"
+            value={selectFilter}
+            data-cy="statusSelect"
+            onChange={(e) => {
+              setSelectFilter(e.target.value as TodoStatus);
+            }}
+          >
+            {Object.values(TodoStatus).map(option => (
+              <option
+                className="is-capitalized"
+                key={option}
+                value={option}
+              >
+                {option}
+              </option>
+            ))}
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          value={filterField}
+          onChange={(event) => setFilterField(event.target.value)}
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
It is better not to disable the linter, but to write the code so that there are no errors.

Reply:
fixed)